common/src/main/java/com/wynntils/wynn/model/territory/TerritoryManager.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.model.territory;
+
+import com.google.common.reflect.TypeToken;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.managers.CoreManager;
+import com.wynntils.core.webapi.WebManager;
+import com.wynntils.core.webapi.request.RequestBuilder;
+import com.wynntils.core.webapi.request.RequestHandler;
+import com.wynntils.mc.event.AdvancementUpdateEvent;
+import com.wynntils.mc.utils.ComponentUtils;
+import com.wynntils.wynn.model.map.poi.Poi;
+import com.wynntils.wynn.model.map.poi.TerritoryPoi;
+import com.wynntils.wynn.model.territory.objects.TerritoryInfo;
+import com.wynntils.wynn.objects.profiles.TerritoryProfile;
+import java.io.File;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import net.minecraft.advancements.Advancement;
+import net.minecraft.advancements.FrameType;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class TerritoryManager extends CoreManager {
+    private static final int TERRITORY_UPDATE_MS = 15000;
+
+    // This is territory POIs as returned by the advancement from Wynncraft
+    private static Map<String, TerritoryPoi> territoryPoiMap = new ConcurrentHashMap<>();
+
+    // This is the profiles as downloaded from Athena
+    private static Map<String, TerritoryProfile> territoryProfileMap = new HashMap<>();
+
+    // This is just a cache of TerritoryPois created for all territoryProfileMap values
+    private static Set<TerritoryPoi> allTerritoryPois = new HashSet<>();
+
+    private static TerritoryUpdateThread territoryUpdateThread;
+
+    public static void init() {
+        territoryUpdateThread = new TerritoryUpdateThread("Territory Update Thread");
+        territoryUpdateThread.start();
+    }
+
+    public static TerritoryProfile getTerritoryProfile(String name) {
+        return territoryProfileMap.get(name);
+    }
+
+    public static Stream<String> getTerritoryNames() {
+        return territoryProfileMap.keySet().stream();
+    }
+
+    public static Set<TerritoryPoi> getTerritoryPois() {
+        return allTerritoryPois;
+    }
+
+    public static List<Poi> getTerritoryPoisFromAdvancement() {
+        // FIXME: Clarify how this differs from getTerritoryPois()...
```

Comment:
So?

Reply:
That was intended as a future reminder to myself. I'll remove it from here and open a bug in GH instead.

Reply:
https://github.com/Wynntils/Artemis/issues/793